src/libmongoc/src/mongoc/mongoc-collection.c

```diff
@@ -3396,6 +3414,10 @@ mongoc_collection_find_and_modify_with_opts (
       bson_append_value (&parts.extra, "hint", 4, &appended_opts.hint);
    }
 
+   if (!bson_empty (&appended_opts.let)) {
+      bson_append_document (&parts.extra, "let", 3, &appended_opts.let);
```

Comment:
While this is consistent with other `mongoc_find_and_modify_appended_opts_t` options, I don't understand why libmongoc appends these to `mongoc_cmd_parts_t` instead of `bson_t command`.

Appending directly to `bson_t command` appears to only be done for the first-class options modeled on `mongoc_find_and_modify_opts_t`.

I suppose it ultimately doesn't make a difference, but is there a reason for this?

On a side note: I suppose `mongoc_find_and_modify_opts_t` only exists for BC reasons and because libmongoc only models the generic `findAndModify` command and not specific CRUD methods (e.g. `findOneAndUpdate`). If these APIs were introduced today, I presume everything would just use the templated options parser, correct?